### PR TITLE
Add get upgrade options and update method for server

### PR DESCRIPTION
**Pull request by mfbmina**
In this PR I added support for getting upgrade options and for update for VMs.

It's needy when you need upgrade cpu or memory of a server.

For usage example, see examples :smile:

### CODE REVIEW

File: lib/fog/softlayer/requests/compute/get_bare_metal_upgrade_item_prices.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    end
+  end
+end
+
+module Fog
+  module Compute
+    class Softlayer
+      class Mock
+        def get_upgrade_item_prices
+          []
```
Comment (fernandes) [inline lib/fog/softlayer/requests/compute/get_bare_metal_upgrade_item_prices.rb]: any reason why its blank? we can put some values here so we are able to simulate using Fog.mock!

**Comment by fernandes** [comment]
lets just check the Mock request then its ok to be merged :+1: :wink:

File: lib/fog/softlayer/requests/compute/get_bare_metal_upgrade_item_prices.rb (hunk 1 above)
Comment (mfbmina) [inline lib/fog/softlayer/requests/compute/get_bare_metal_upgrade_item_prices.rb]: no good reason. I was just waiting for a BM provision and I forgotten to add a mock here. Fixing it!

Comment (fernandes) [inline lib/fog/softlayer/requests/compute/get_bare_metal_upgrade_item_prices.rb]: great! :fireworks:

**Comment by mfbmina** [comment]
@fernandes , updated here! please take a look!

**Comment by fernandes** [comment]
why raising for BM request, not available via API or just not implemented?

**Comment by fernandes** [comment]
this one needs rebase after https://github.com/fog/fog-softlayer/pull/57 was merged

**Comment by mfbmina** [comment]
Just not implemented, because I'm having some issues with this feature and BM's.

**Comment by mfbmina** [comment]
@fernandes rebased done!

**Pull request merged**